paasta_tools/cli/cmds/validate.py

```diff
@@ -209,14 +209,17 @@ def validate_chronos(service_path):
     return returncode
 
 
-def paasta_validate(args):
+def paasta_validate(args, service_path=None):
     """Analyze the service in the PWD to determine if conf files are all valid
 
     :param args: argparse.Namespace obj created from sys.args by cli
     """
 
-    service = args.service
-    soa_dir = args.yelpsoa_config_root
+    if service_path:
```

Comment:
I don't really like this kind of... thing.

Can you just make paasta_validate take in args and process them..

And then make a `paasta_validate_soa_configs` that *just takes a soa_configs* input?

Then paasta_validate_soa_configs becomes much more like what you might expect, it takes a service to validate and a soa_configs path to look at where to validate.